### PR TITLE
nixos/installation-cd-base: add git & rsync

### LINKED ISSUE #325170: minimal iso does not include git
### Describe the bug
The nixos minimal iso doesn't include `git`, which causes `nixos-install` to fail when fetching a remote flake source or any `fetchgit`  calls within a configuration.

### Steps To Reproduce
1. Boot minimal installation media
2. Proceed with standard installation steps until nixos-install
3. `nixos-install --flake github:user/repo#host` fails

### Expected behavior
`git` should be included in the minimal iso. For a distro that prides itself on being reproducible from text files, it feels like unnecessary friction having to `nix-shell` for something as basic and ubiquitous as git.

### Additional context
This has been brought up before in #78127, but was rejected [after a discussion in nixos-dev](https://logs.nix.samueldr.com/nixos-dev/2020-01-21#2991326). Maybe it's time to revisit this decision?

### Notify maintainers
N/A

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by JerrySM64**
## Description of changes
Add git and rsync to installation-cd-base so the minimal iso also has those tools out of the box. Fixes #325170

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Frontear — APPROVED** [review]
This would be a great change. Approved!

**Comment by Gerg-L** [comment]
should `gitMinimal` be used instead?
it's a third the closure size
```bash
$ nix path-info -Sh $(nix build --print-out-paths nixpkgs#git)
/nix/store/1zyjd4qmdgwrkvcvmxddrka6lzzx1h2b-git-2.44.1   316.3 MiB
$ nix path-info -Sh $(nix build --print-out-paths nixpkgs#gitMinimal)
/nix/store/vjzkh497v13asja08g41zilix8lnfsnq-git-minimal-2.44.1   114.7 MiB
```
the only downside i can see is that it doesn't include manual pages

**Comment by Gerg-L** [comment]
Tested with rsync + git/gitMinimal on nixos-unstable like so:
```
cd nixos
nix-build -A config.system.build.isoImage -I nixos-config=modules/installer/cd-dvd/installation-cd-minimal.nix default.nix
du -h $(readlink ./result)
```
baseline: 1052810

git+rysnc: 1072462
gitMinimal+rsync: 1068182
git only: 1072098
rsync only: 1014754

ok so git vs gitMinimal does not make a significant difference

I would also remove this line in this PR
 
<https://github.com/NixOS/nixpkgs/blob/67ab30a1aac28c793dc119a74ceab95643a330f8/nixos/modules/installer/cd-dvd/installation-cd-graphical-base.nix#L59-L61>
other than that looks good

**Comment by JerrySM64** [comment]
Will squash in a moment. The Web IDE doesn't let me do it.

**Review by NotAShelf — APPROVED** [review]
LGTM

**Comment by JerrySM64** [comment]
Do I still need to squash? That LGTM confused me a bit.

**Comment by NotAShelf** [comment]
Yes you still need to squash, I didn't notice the history sorry.

**Comment by JerrySM64** [comment]
Alright, it is now squashed. Took a bit, because git was acting up with the rebase command, sorry.

**Review by Gerg-L — APPROVED** [review]
LGTM

File: nixos/modules/installer/cd-dvd/installation-cd-base.nix [hunk 1]
```diff
@@ -46,5 +46,11 @@ with lib;
     done
   '';
 
+  environment.defaultPackages = with pkgs; [
+    # Include some version control tools.
+    git
+    rsync
+  ];
```
Comment (Pandapip1) [inline nixos/modules/installer/cd-dvd/installation-cd-base.nix:53]: Nit:

```suggestion
  environment.defaultPackages = with pkgs; [
    rsync
  ];

  programs.git.enable = true;
```

This way, the git package that is installed can be configured, and setting `programs.git.package` won't result in two `git`s being in `defaultPackages`.

Comment (Gerg-L) [reply nixos/modules/installer/cd-dvd/installation-cd-base.nix:53]: fair points, but if you're doing that it should probably `mkDefault`'d

Comment (Pandapip1) [reply nixos/modules/installer/cd-dvd/installation-cd-base.nix:53]: > fair points, but if you're doing that it should probably mkDefault'd

Updated

Comment (JerrySM64) [reply nixos/modules/installer/cd-dvd/installation-cd-base.nix]: Updated it.

**Review by Gerg-L — APPROVED** [review]

**Review by Pandapip1 — APPROVED** [review]
Diff lgtm

**Comment by JerrySM64** [comment]
Is it going to be merged or do we hold it for the other possible solutions mentioned in the issue I mentioned in the beginning? Not trying to be rude or impatient, just curious.

**Comment by Gerg-L** [comment]
Will probably be merged whenever someone with merged access glaces at it

File: nixos/modules/installer/cd-dvd/installation-cd-base.nix [hunk 2]
```diff
@@ -46,5 +46,12 @@ with lib;
     done
   '';
 
+  environment.defaultPackages = with pkgs; [
+    # Include some version control tools.
```
Comment (Frontear) [inline nixos/modules/installer/cd-dvd/installation-cd-base.nix:50]: very much a nitpick but remove that comment, otherwise lgtm

**Review by Frontear — APPROVED** [review]

File: nixos/modules/installer/cd-dvd/installation-cd-base.nix (hunk 2 above)
Comment (JerrySM64) [reply nixos/modules/installer/cd-dvd/installation-cd-base.nix:50]: Will do that later.

Comment (JerrySM64) [reply nixos/modules/installer/cd-dvd/installation-cd-base.nix]: Done

**Review by Gerg-L — APPROVED** [review]
LGTM

**Review by iynaix — APPROVED** [review]

**Review by luckymachi — APPROVED** [review]
LGTM!

**Review by myclevorname — APPROVED** [review]
LGTM

**Comment by myclevorname** [comment]
> Will probably be merged whenever someone with merged access glaces at it

@Aleksanaa please

**Review by LilleAila — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**